lib/helper/Appium.js

```diff
@@ -0,0 +1,839 @@
+            if (res) {
+              return true;
+            } else {
+              return browser[direction](scrollLocator, offset, speed).getSource().then((source) => {
+                if (source === currentSource) {
+                  err = true
+                } else {
+                  currentSource = source;
+                  return false;
+                }
+              });
+            }
+          })
+      }
+    }, timeout * 1000)
+  }
+
+  /**
+   * Perform a swipe up on an element.
+   * Appium: support Android and iOS
+   */
+  swipeUp(locator, yOffset, speed) {
+    if (this.options.desiredCapabilities.platformName) {
+      return this.browser.swipeUp(withStrictLocator(locator), yOffset, speed);
+    } else
+      throw new Error(`swipeUp method can be used only with apps`);
+  }
+
+  /**
+   * Switch to the specified context
+   * Appium: support only Android
+   */
+  switchToContext(value) {
+    if (this.options.desiredCapabilities.platformName) {
+      return this.browser.context(value);
+    } else
+      throw new Error(`switchToContext method can be used only with apps`);
+  }
+
+  /**
+   * Switch the state (enabled/disabled) of airplane mode.
+   * Appium: support only Android
+   */
+  toggleAirplaneMode() {
+    if (this.options.desiredCapabilities.platformName == "Android") {
+      return this.browser.toggleAirplaneMode();
+    } else
+      throw new Error(`toggleAirplaneMode method can be used only with Android apps`);
+  }
+
+  /**
+   * Switch the state (enabled/disabled) of the location service.
+   * Appium: support only Android
+   */
+  toggleLocationServices() {
+    if (this.options.desiredCapabilities.platformName == "Android") {
+      return this.browser.toggleLocationServices();
+    } else
+      throw new Error(`toggleLocationServices method can be used only with Android apps`);
+  }
+
+  /**
+   * Switch the state (enabled/disabled) of the wifi service.
+   * Appium: support only Android
+   */
+  toggleWiFi() {
+    if (this.options.desiredCapabilities.platformName == "Android") {
+      return this.browser.toggleWiFi();
+    } else
+      throw new Error(`toggleWiFi method can be used only with Android apps`);
+  }
+
+  /**
+   * Performs a specific touch action. The action object need to contain the action name, x/y coordinates
+   * Appium: support Android and iOS
+   */
+  touchPerform(actions) {
+    if (this.options.desiredCapabilities.platformName) {
+      return this.browser.touchPerform(actions);
+    } else
+      throw new Error(`touchPerform method can be used only with apps`);
+  }
+
+  /**
+   * Unlock the device.
+   * Appium: support only Android
+   */
+  unlockDevice() {
+    if (this.options.desiredCapabilities.platformName) {
+      return this.browser.unlock();
+    } else
+      throw new Error(`unlockDevice method can be used only with Android apps`);
+  }
+
+  /*
+   * End Appium Methods
+   */
+
+}
+
+function isCSSorXPathLocator(locator) {
```

Comment:
too bad you had to copy-paste this part... 
We need to think for a way we can reuse that.